app/src/main/java/org/sagebionetworks/research/mpower/tracking/TrackingTabFragment.java

```diff
@@ -146,11 +196,124 @@ private void setupActionBar(@Nullable StudyBurstItem item) {
+    private void showActionBarFlow(@Nonnull StudyBurstItem item) {
+        ScheduledActivityEntity nextCompletionTask = item.getNextCompletionActivityToShow();
+        if (!item.hasCompletedMotivationSurvey() && item.getMotivationSurvey() != null) {
+            launchRsSurvey(item.getMotivationSurvey());
+        } else if (nextCompletionTask != null) {
+            launchRsSurvey(nextCompletionTask);
+        } else {
+            goToStudyBurst();
+        }
+    }
+
+    /**
+     * Launches an old style ResearchStack SmartSurveyTask
+     * @param surveySchedule of the survey to launch
+     */
+    private void launchRsSurvey(@Nullable ScheduledActivityEntity surveySchedule) {
+        if (surveySchedule == null ||
+                surveySchedule.getActivity() == null ||
+                surveySchedule.getActivity().getSurvey() == null) {
+            return; // NPE guard statements
+        }
+        hasShownStudyBurst = true;
+        trackingStatusBar.setEnabled(false);
+        trackingStatusBar.setProgressBarVisibility(View.VISIBLE);
+        currentSurveySchedule = surveySchedule;
+        // This triggers a SingleLiveEvent which only posts a success once
+        studyBurstViewModel.loadRsSurvey(surveySchedule);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode == Activity.RESULT_OK && requestCode == REQUEST_TASK) {
+            TaskResult taskResult = (TaskResult)
+                    data.getSerializableExtra(ViewTaskActivity.EXTRA_TASK_RESULT);
+
+            if (currentSurveyTask != null) {
+                // This will trigger any after-rule processing like adding data groups based on survey answers
+                currentSurveyTask.processTaskResult(taskResult);
+            }
+            if (currentSurveySchedule != null) {
+                currentSurveySchedule.setStartedOn(Instant.ofEpochMilli(taskResult.getStartDate().getTime()));
+                currentSurveySchedule.setFinishedOn(Instant.ofEpochMilli(taskResult.getEndDate().getTime()));
+                // This function updates the schedule on bridge and in the ScheduleRepository
+                studyBurstViewModel.updateScheduleToBridge(currentSurveySchedule);
+                // This function uploads the result of the task to S3
+                studyBurstViewModel.uploadResearchStackTaskResultToS3(currentSurveySchedule, taskResult);
+
+                if (MOTIVATION.equals(currentSurveySchedule.activityIdentifier())) {
+                    // send the user straight into the study burst
+                    goToStudyBurst();
+                }
+            }
+        }
+        currentSurveyTask = null;
+        currentSurveySchedule = null;
+
+        // Check this at the end because it may set currentSurveyTask and currentSurveySchedule
+        if (resultCode == Activity.RESULT_OK &&
+                requestCode == StudyBurstActivity.Companion.getREQUEST_CODE_STUDY_BURST()) {

```

Comment:
If you do `const val REQUEST_CODE_STUDY_BURST` then in Java you don't need to specify the Companion and you can access it as a field instead of get/set